torch/_inductor/scheduler.py

```diff
@@ -1250,7 +1254,30 @@ def fuse(
     ) -> FusedSchedulerNode:
         assert node1.scheduler is node2.scheduler
         assert isinstance(node1, (SchedulerNode, FusedSchedulerNode))
-        assert isinstance(node2, (SchedulerNode, FusedSchedulerNode))
+        if node1.is_template() and isinstance(node2, ExternKernelSchedulerNode):
+            # Fuse multi outputs template and its outputs
+            #   * Node1 has memorydep of MultiOutput in reads
+            #   * Node2 has StarDep of MultiOutput in writes
+            # Rewrite the Node2' StarDep to MemoryDep
```

Comment:
Elaborate why we do this in the comment?:

Reply:
Thanks for the comment. Add the explanation for why we need to change the type of dependency.